### ISSUE TITLE
Weird domain can't be resolved by the CLI client

### ISSUE THREAD

**Issue by CrendKing** [nextdns/nextdns#347]
I have this one specific domain that can't be resolved just by the CLI client. I verified with the log from website that this domain is resolvable. It seems to be a client side problem. And it only happens on this particular domain (so far I only found this one).

1. I'm on Windows. When I do `nslookup <the_domain>`, I get
```
Server:  RT-AC68R
Address:  192.168.1.1

*** No internal type for both IPv4 and IPv6 Addresses (A+AAAA) records available for <the_domain>
```
Wireshark also confirms that no answer was replied for those queries.
2. If I `nslookup <the_domain> 1.1.1.1`, I get the IPs.
3. Now I ssh to my router. If I `nslookup <the_domain> 192.168.1.1` I get
```
Server:    192.168.1.1
Address 1: 192.168.1.1 RT-AC68R.

nslookup: can't resolve '<the_domain>'
```
4. If I `nslookup <the_domain>`, I get
```
Server:    127.0.1.1
Address 1: 127.0.1.1

Name:      <the_domain>
Address 1: <IP 1>
Address 2: <IP 2>
...
```
5. If I go back to step 3, I still get the same no IP response. Cache does not work here.
6. I enabled `-log-queries`. I can see entries for queries in step 1 and 3:
```
Oct 16 11:22:07 nextdns[7450]: Query 192.168.1.1 UDP A <the_domain>. (qry=70/res=70) 297ms HTTP/2.0
Oct 16 11:22:08 nextdns[7450]: Query 192.168.1.100 UDP A <the_domain>. (qry=80/res=70) 15ms HTTP/2.0
```
But queries from step 4 do not appear there.
7. I repeated step 4 with `dig`. Same result, getting IPs.
8. What's weird is, if I dig with local server `dig <the_domain> @192.168.1.1`, not only I get a whole bunch more IPs, but it caches so if I go back to step 1 and 3, I also get IPs. You can see the difference in the log:
```
Oct 16 11:26:39 nextdns[7450]: Query 192.168.1.1 UDP A <the_domain>. (qry=82/res=1414) 769ms HTTP/2.0
```
8.5. After the cache expires, step 1 and 3 go back to the no IP response again.
9. If I disable the CLI client and go back to stubby, the problem goes away.

**Context**
 - Version [e.g. 1.5.2]: NextDNS 1.8.6, but I remember this problem has existed since the debut.
 - Platform [e.g. macOS, ASUS-Merlin]: Windows 10, ASUS-Merlin AC-68R

**Comment by rs** [comment]
I would need <the_domain> to reproduce.

**Comment by CrendKing** [comment]
I'm afraid I can't disclose the domain name. However, after hours of bisecting all the possible combinations of configs and options, I found the culprit! It is the "Anonymized EDNS Client Subnet" in the Performance section. I created a config with everything disabled, and no blocklist, then enable thing one by one. That feature is what's causing the issue. Once I turn off that in my original config, it worked.

I don't know why does that feature conflict with the particular domain and only on the client. Feel free to close the ticket.
